src/util/time_test.sh

```diff
@@ -22,14 +22,41 @@ function test_now_millis() {
 }
 readonly -f test_now_millis
 
-function test_time_now_day() {
+function test_time_now_day_of_week() {
         local day
-        day=$(time_now_day) || \
+        day=$(time_now_day_of_week) || \
                 assert_fail
         assert_gt "${day}" 0
         assert_lt "${day}" 8
 }
-readonly -f test_time_now_day
+readonly -f test_time_now_day_of_week
+
+function test_time_now_day_of_week_str() {
+        local day
+        day=$(time_now_day_of_week_str) || \
+                assert_fail
+
+        case "${day}" in
+        "Monday"|"monday") ;;
+        "Tuesday"|"tuesday") ;;
+        "Wednesday"|"wednesday") ;;
+        "Thursday"|"thursday") ;;
+        "Friday"|"friday") ;;
+        "Saturday"|"saturday") ;;
+        "Sunday"|"sunday") ;;
+        *) assert_fail "non-existing day of the week";;
+        esac
+}
+readonly -f test_time_now_day_of_week_str
+
+function test_time_now_day_of_month() {
+        local day
+        day=$(time_now_day_of_month) || \
+                assert_fail
+        assert_gt "${day}" 0
```

Comment:
the only change I would make is on this line and the follow line
```
[ "${day}" -lt 0 ] && assert_fail
[ "${day}" -gt 32 ] && assert_fail
return 0
```

Reply:
I can add it, but why if we have `assert_gt` and `assert_lt` methods, and the whole file uses these methods, for example in the function [`test_time_now_day_of_week`](https://github.com/IvanGrigorik/gobash/blob/e9945ba9dcd57307806a32b23b07b3c56193ae02/src/util/time_test.sh#L25-L30) function?

Maybe then I should change all occurrences in this file?

Reply:
Great point and question.
`assert_` functions are legacy code before we decided to make our library fully compatible with `set -e` (i.e., if invokes test directly as a function the execution should not stop). Documented here: https://engineeringsoftware.github.io/gobash/testing.html#asserting-results

You are correct that all those assert should be changed in this file. If you have time, go ahead. If not, you can only update your own change. Thanks for the contribution in either case.

Reply:
Updated.